core/src/packet.rs

```diff
@@ -130,8 +151,10 @@ where
 {
     fn drop(&mut self) {
         unsafe {
-            U::from_raw_const_ptr(self.raw().user_data.cast_const().cast());
-            sys::tb_client_release_packet(self.handle.raw, self.raw);
+            drop(U::from_raw_const_ptr(
+                self.raw().user_data.cast_const().cast(),
+            ));
+            drop(Box::from_raw(self.raw));
```

Comment:
@zetanumbers We do `drop` here because [allocated packet previously](https://github.com/tigerbeetle-rust/tigerbeetle-unofficial/blob/1c4c443561ae84d3f27047a216a813dfa02d4ccf/core/src/packet.rs#L39). Can you please provide more details why we shouldn't do it?

Reply:
Objects are dropped automatically at the end of the expression.

Reply:
@50U10FCA7 recreated `Box` will be automatically dropped at the end of the scope.

@zetanumbers however, putting an explicit `drop()` call here emphasizes what's going on, which may be unclear for some readers. I'd like to keep it.

Reply:
@zetanumbers Do you want to use `_ = Box::from_raw(self.raw);` instead?

<img width="508" alt="Снимок экрана 2024-11-11 в 06 12 15" src="https://github.com/user-attachments/assets/58671dfd-15b2-4e1e-9527-dac0be0a96a9">

Reply:
Sure